### PR TITLE
added option for High DPI displays to enable and disable scaling 

### PR DESCRIPTION
**Pull request by yjeanrenaud**
https://github.com/nextcloud/desktop/issues/5373  
So I added a checkbox to the settings and the corresponding code to handle that Qt function 

Signed-off-by: Yves Jeanrenaud <yves.jeanrenaud@gmail.com>

### CODE REVIEW

File: src/libsync/configfile.cpp [hunk 1]
```diff
@@ -236,7 +236,30 @@ void ConfigFile::setShowInExplorerNavigationPane(bool show)
...
+    #if QTLEGACY
+        (QSysInfo::windowsVersion() < QSysInfo::WV_WINDOWS10);
+    #else
+        QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows10;
+    #endif
+    //TODO need to check for platform windows:dpiawareness
+#else
+        false
+#endif
+        ;
```
Comment (claucambra) [inline src/libsync/configfile.cpp:252]:
```suggestion
        false;
#endif
```

File: src/libsync/configfile.cpp [hunk 2]
```diff
@@ -236,7 +236,30 @@ void ConfigFile::setShowInExplorerNavigationPane(bool show)
...
+bool ConfigFile::scaleForHighDPIDisplay() const
+{
+    const bool defaultValue =
+#ifdef Q_OS_WIN
+    #if QTLEGACY
+        (QSysInfo::windowsVersion() < QSysInfo::WV_WINDOWS10);
+    #else
+        QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows10;
+    #endif
+    //TODO need to check for platform windows:dpiawareness
```
Comment (claucambra) [inline src/libsync/configfile.cpp:248]: Please implement this TODO

File: src/gui/main.cpp [hunk 3]
```diff
@@ -87,6 +87,15 @@ int main(int argc, char **argv)
     if (app.devicePixelRatio() > 1)
         QApplication::setStyle(QStringLiteral("fusion"));
 #endif // Q_OS_WIN
+    
+    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
+#ifdef Q_OS_WIN
+    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+#endif // !Q_OS_WIN
+    if(_configFile->scaleForHighDPIDisplay->scale == true) {
```
Comment (claucambra) [inline src/gui/main.cpp:95]:
```suggestion
    if (_configFile->scaleForHighDPIDisplay->scale) {
```

**Review by claucambra — CHANGES_REQUESTED** [review]
A couple of comments :)

**Comment by claucambra** [comment]
Could you also add a screenshot of the settings window with your changes?

**Comment by alphasixtyfive** [comment]
Would love to see it merged

**Comment by Julfried** [comment]
Any update on merging this?

**Comment by Rello** [comment]
Hello,

Thank you for your contribution to the Desktop Client with this pull request.
We are closing this request as it is outdated, no longer relevant (e.g., due to Qt 6), or does not align with the current roadmap.

We truly value your support and hope to see more contributions from you in the future!


Best regards,
Nextcloud Desktop Team

**Pull request closed without merging**